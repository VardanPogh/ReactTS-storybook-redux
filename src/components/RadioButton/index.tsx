import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Radio from '@material-ui/core/Radio'
import RadioGroup from '@material-ui/core/RadioGroup'
import FormControlLabel from '@material-ui/core/FormControlLabel'

const useStyles = makeStyles({
  radio: {
    color: '#014bad !important',
    borderColor: '#014bad !important',
    '&checked': {
      color: '#014bad !important',
      borderColor: '#014bad !important',
    },
  },
})

interface Items {
  value: string
  label: string
}

interface Props {
  items: Items[]
  defaulValue?: string
  labelPlacement: 'bottom' | 'end' | 'start' | 'top'
  onChange?(e: any): void
}

const RadioButtonGroup: React.FC<Props> = ({
  items,
  defaulValue,
  labelPlacement,
  onChange,
}) => {
  const classes = useStyles()
  return (
    <RadioGroup
      row
      aria-label="radio-group"
      name="radio-group"
      defaultValue={defaulValue}
    >
      {items.map((item, index) => (
        <FormControlLabel
          key={index}
          value={item.value}
          control={<Radio className={classes.radio} />}
          label={item.label}
          labelPlacement={labelPlacement}
          onChange={onChange}
        />
      ))}
    </RadioGroup>
  )
}

export default RadioButtonGroup
