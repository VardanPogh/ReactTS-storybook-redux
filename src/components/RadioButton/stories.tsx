import React from 'react'
import { withKnobs, text, object, select } from '@storybook/addon-knobs'

import RadioButton from '.'

export default {
  title: 'RadioButton',
  component: RadioButton,
  decorators: [withKnobs],
}

const labelPlacement = {
  Bottom: 'bottom',
  End: 'end',
  Start: 'start',
  Top: 'top',
}

const items = [
  {
    value: 'yes',
    label: 'Yes',
  },
  {
    value: 'no',
    label: 'No',
  },
  {
    value: 'maybelater',
    label: 'Maybe Later',
  },
]

export const Default: React.FC = () => (
  <RadioButton
    items={object('Radio', items)}
    defaulValue={text('Default Value', 'yes')}
    // eslint-disable-next-line
    // @ts-ignore
    labelPlacement={select(
      'labelPlacement',
      labelPlacement,
      labelPlacement.Bottom
    )}
  />
)
