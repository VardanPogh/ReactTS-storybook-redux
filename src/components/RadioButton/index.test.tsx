import React from 'react'
import { render } from '@testing-library/react'
import RadioButton from '.'

const items = [
  {
    value: 'yes',
    label: 'Yes',
  },
  {
    value: 'no',
    label: 'No',
  },
  {
    value: 'maybelater',
    label: 'Maybe Later',
  },
]
describe('RadioButton Test', () => {
  test('Loads RadioButton', () => {
    const { container } = render(
      <RadioButton items={items} defaulValue="yes" labelPlacement="bottom" />
    )
    expect(container).toMatchSnapshot()
  })
})
