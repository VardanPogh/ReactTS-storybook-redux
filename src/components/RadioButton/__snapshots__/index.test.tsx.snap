// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Test Loads RadioButton 1`] = `
<div>
  <div
    aria-label="radio-group"
    class="MuiFormGroup-root MuiFormGroup-row"
    role="radiogroup"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementBottom"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiRadio-root MuiRadio-colorSecondary makeStyles-radio-1 PrivateSwitchBase-checked-3 Mui-checked MuiIconButton-colorSecondary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input-5"
            name="radio-group"
            type="radio"
            value="yes"
          />
          <div
            class="PrivateRadioButtonIcon-root-6 PrivateRadioButtonIcon-checked-8"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-7"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Yes
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementBottom"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiRadio-root MuiRadio-colorSecondary makeStyles-radio-1 MuiIconButton-colorSecondary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-5"
            name="radio-group"
            type="radio"
            value="no"
          />
          <div
            class="PrivateRadioButtonIcon-root-6"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-7"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        No
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementBottom"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiRadio-root MuiRadio-colorSecondary makeStyles-radio-1 MuiIconButton-colorSecondary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-5"
            name="radio-group"
            type="radio"
            value="maybelater"
          />
          <div
            class="PrivateRadioButtonIcon-root-6"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-7"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Maybe Later
      </span>
    </label>
  </div>
</div>
`;
